doyensec/detectors/grafana_zabbix_credential_disclosure/src/test/java/com/google/tsunami/plugins/detectors/cve/cve202226148/CredentialsDisclosureInGrafanaZabbixIntegrationDetectorTest.java

```diff
@@ -0,0 +1,157 @@
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.Resources;
+import com.google.inject.Guice;
+import com.google.tsunami.common.net.http.HttpClientModule;
+import com.google.tsunami.common.net.http.HttpStatus;
+import com.google.tsunami.common.time.testing.FakeUtcClock;
+import com.google.tsunami.common.time.testing.FakeUtcClockModule;
+import com.google.tsunami.proto.DetectionReportList;
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.TargetInfo;
+import java.io.IOException;
+import java.time.Instant;
+import javax.inject.Inject;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link CredentialsDisclosureInGrafanaZabbixIntegrationDetector}. */
+@RunWith(JUnit4.class)
+public final class CredentialsDisclosureInGrafanaZabbixIntegrationDetectorTest {
+
+  private final FakeUtcClock fakeUtcClock =
+      FakeUtcClock.create().setNow(Instant.parse("2020-01-01T00:00:00.00Z"));
+
+  @Inject private CredentialsDisclosureInGrafanaZabbixIntegrationDetector detector;
+
+  private MockWebServer mockGrafanaService;
+
+  private final String disclosedCredentialGrafanaLoginResponse;
+  private final String secureGrafanaLoginResponse;
+  private final String disclosedCredentialGrafanaWelcomeResponse;
+  private final String secureGrafanaWelcomeResponse;
+
+  public CredentialsDisclosureInGrafanaZabbixIntegrationDetectorTest() throws IOException {
+    this.disclosedCredentialGrafanaLoginResponse =
+        Resources.toString(
+            Resources.getResource(this.getClass(), "disclosedCredentialGrafanaLoginResponse.html"),
+            UTF_8);
+    this.secureGrafanaLoginResponse =
+        Resources.toString(
+            Resources.getResource(this.getClass(), "secureGrafanaLoginResponse.html"), UTF_8);
+    this.disclosedCredentialGrafanaWelcomeResponse =
+        Resources.toString(
+            Resources.getResource(
+                this.getClass(), "disclosedCredentialGrafanaWelcomeResponse.html"),
+            UTF_8);
+    this.secureGrafanaWelcomeResponse =
+        Resources.toString(
+            Resources.getResource(this.getClass(), "secureGrafanaWelcomeResponse.html"), UTF_8);
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    mockGrafanaService = new MockWebServer();
+    mockGrafanaService.start();
+
+    Guice.createInjector(
+            new FakeUtcClockModule(fakeUtcClock),
+            new HttpClientModule.Builder().build(),
+            new CredentialsDisclosureInGrafanaZabbixIntegrationDetectorBootstrapModule())
+        .injectMembers(this);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    mockGrafanaService.shutdown();
+  }
+
+  @Test
+  public void detect_whenVulnerable_reportsVulnerability()
```

Comment:
The test is currently performing 4 consecutive requests and verifying that the report contains the scanned target. This presents multiple issues:

1. If any of the mocked requests is not vulnerable, the test will still pass, resulting in a false positive. This can be verified by simply changing one of the mocked requests to:
```java
mockGrafanaService.enqueue(
    new MockResponse()
        .setResponseCode(HttpStatus.OK.code())
        .setBody(secureGrafanaLoginResponse));
```

2. The test mocks 4 requests in an attempt to verify that all vulnerable [endpoints](https://github.com/google/tsunami-security-scanner-plugins/blob/681ad43391473770c702b40a6630684ec4127183/doyensec/detectors/grafana_zabbix_credential_disclosure/src/main/java/com/google/tsunami/plugins/detectors/cve/cve202226148/CredentialsDisclosureInGrafanaZabbixIntegrationDetector.java#L155) are covered. However it does not verify that the detector actually invoked them. The mocks are generic and will be returned for any URL.

With regards to 1) I'd recommend verifying that the number of instances of the target service in the report is equal to the number of mocked requests. This will ensure that all mocked requests did in fact are vulnerable.

Regarding 2), consider adding a custom `Dispatcher` such as the one defined [here](https://github.com/google/tsunami-security-scanner-plugins/blob/master/community/detectors/apache_http_server_cve_2021_41773/src/test/java/com/google/tsunami/plugins/detectors/cve202125646/ApacheHttpServerCVE202141773VulnDetectorTest.java#L172), that will return a request only for the expected endpoints. Then you can verify the number of expected request with:
```java
assertThat(mockWebServer.getRequestCount()).isEqualTo(<EXPECTED_COUNT>);
```
This applies for both test.